client/homebrew/editor/editor.jsx

```diff
@@ -174,9 +174,39 @@ const Editor = createClass({
 		}
 	},
 
-	brewJump : function(){
-		const currentPage = this.getCurrentPage();
-		window.location.hash = `p${currentPage}`;
+	brewJump : function(targetPage=this.getCurrentPage()){
+		if(!window || this.isMeta()) return;
+		window.frames['BrewRenderer'].contentDocument.getElementById(`p${targetPage}`).scrollIntoView({ behaviour: 'auto', block: 'start' });
+		// const hashPage = (page != 1) ? `p${page}` : '';
+		// window.location.hash = hashPage;
+	},
+
+	sourceJump : function(targetLine=null){
+		if(this.isText() || this.isStyle()) {
+			if(targetLine == null) {
+				const brewPosition = window.frames['BrewRenderer'].contentDocument.getElementsByClassName('brewRenderer').item(0).scrollTop;
```

Comment:
We do a similar calculation for this in `the `handleScroll()` function in `brewRenderer.jsx` but it's actually pretty simple:

```
const target = e.target;
viewablePageNumber = Math.floor(target.scrollTop / target.scrollHeight * prevState.pages.length)
```

Where `e` is the `brewRenderer` DIV element.